### PR TITLE
BUG: Update pyproject.toml with minimum Python version of 3.8.

### LINKED ISSUE #2858: 5.0.0 should not be pip installable for Python < 3.8
`pypdf 5.0.0` is pip-installable for Python 3.7, but support for 3.7 is dropped as noted in the [Changelog](https://pypdf.readthedocs.io/en/stable/meta/CHANGELOG.html#version-5-0-0-2024-09-15).

Packages that require `pypdf > X` but don't specify `<5` are subject to erroneously installing `pypdf >= 5` when running on Python 3.7.

## Environment

Python 3.7, any OS

```bash
$ python -m platform
Windows-10-10.0.22621-SP0

$ python -c "import pypdf;print(pypdf._debug_versions)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\__init__.py", line 11, in <
module>
    from ._doc_common import DocumentInformation
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\_doc_common.py", line 48, i
n <module>
    from ._encryption import Encryption
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\_encryption.py", line 46, i
n <module>
    from ._utils import logger_warning
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\_utils.py", line 523, in <m
odule>
    @dataclass
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\_utils.py", line 525, in Fi
le
    from .generic import IndirectObject
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\generic\__init__.py", line
38, in <module>
    from ._base import (
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\generic\_base.py", line 37,
 in <module>
    from .._protocols import PdfObjectProtocol, PdfWriterProtocol
  File "C:\Users\zmeve\AppData\Local\Programs\Python\Python37\lib\site-packages\pypdf\_protocols.py", line 5, in
<module>
    from typing import IO, Any, Dict, List, Optional, Protocol, Tuple, Union
ImportError: cannot import name 'Protocol' from 'typing' (C:\Users\zmeve\AppData\Local\Programs\Python\Python37\l
ib\typing.py)
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
import pypdf
print(pypdf.__version__)
```

## Traceback

Same as listed above.

## Suggested Fixes

1. Update the `pyproject.toml` to mark the minimum Python version as 3.8
2. "Yank" the `pypdf 5.0.0` distribution from PyPI for Python 3.7. I'm not sure if this is possible for a specific Python version - if not, it may still be best to yank the entire 5.0.0 release and re-publish it as 5.0.0.post1, or even 5.0.1.

### PR DESCRIPTION
**Pull request by zmeves**
Closes #2858.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -53,7 +53,7 @@ persistent=yes
 
 # Minimum Python version to use for version dependent checks. Will default to
 # the version used to run pylint.
-py-version=3.7
+py-version=3.8
```
Comment (pubpub-zz) [inline .pylintrc:56]: is `.pylintrc` still required ? (I'm not an expert)

File: pyproject.toml [hunk 2]
```diff
@@ -41,13 +40,11 @@ Changelog = "https://pypdf.readthedocs.io/en/latest/meta/CHANGELOG.html"
...
 full = [
-    "cryptography; python_version >= '3.7'",
-    "PyCryptodome; python_version == '3.6'",
+    "cryptography; python_version >= '3.8'",
     "Pillow>=8.0.0",
 ]
 crypto = [
-    "cryptography; python_version >= '3.7'",
-    "PyCryptodome; python_version == '3.6'",
+    "cryptography; python_version >= '3.8'",
```
Comment (pubpub-zz) [inline pyproject.toml:47]: PyCryptoDome should be proposed as an `crypto_alternative`

File: .pylintrc (hunk 1 above)
Comment (stefan6419846) [reply .pylintrc:56]: No, it is not due to introducing `ruff`.

Comment (zmeves) [reply .pylintrc:56]: It seems simplest to remove this file then

Comment (pubpub-zz) [reply .pylintrc:56]: Good to me

Comment (zmeves) [reply .pylintrc]: Resolved via 6a5d10c794099fb5ba2974e171987c5af956baf6

File: pyproject.toml (hunk 2 above)
Comment (zmeves) [reply pyproject.toml:47]: Is that change within the scope of this MR? I have no input on `cryptography` vs `PyCryptodome`

Comment (pubpub-zz) [reply pyproject.toml:47]: Let's keep it on side for the moment.

Comment (stefan6419846) [reply pyproject.toml:47]: This is out of scope for now as it might need further discussions.

**Comment by pubpub-zz** [comment]
@stefan6419846 OK for you ?

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**